packages/lib/slots.ts

```diff
@@ -192,18 +195,22 @@ function buildSlotsWithDateRanges({
       ? range.start
       : startTimeWithMinNotice;
 
-    slotStartTime =
-      slotStartTime.minute() % interval !== 0
-        ? slotStartTime.startOf("hour").add(Math.ceil(slotStartTime.minute() / interval) * interval, "minute")
-        : slotStartTime;
-
     // Adding 1 minute to date ranges that end at midnight to ensure that the last slot is included
     const rangeEnd = range.end
       .add(dayjs().tz(organizerTimeZone).utcOffset(), "minutes")
       .isSame(range.end.endOf("day").add(dayjs().tz(organizerTimeZone).utcOffset(), "minutes"), "minute")
       ? range.end.add(1, "minute")
       : range.end;
 
+    slotStartTime =
+      slotStartTime.minute() % interval !== 0
+        ? showOptimizedSlots
+          ? slotStartTime.add(rangeEnd.diff(slotStartTime, "minutes") % interval, "minute")
```

Comment:
This ensures within a time range, always maximum available (optimum) slots  are shown (without loss of any slot) And also Start of the Hour is respected.
Also , this happens only if user availability starts in minutes , say 10:10 , 10:05 (instead of 10:00 normally) and user has chosen to optimize.
For all other scenarios, the existing behaviour applies and i understand that is expected.